syntax = "proto3";

package observability;

// Observability configuration for tracing, metrics, and logging
message Config {
    // Service identification
    string service_name = 1;
    string service_version = 2;
    string environment = 3;
    
    // OTLP exporter
    string otlp_endpoint = 4;
    
    // Feature toggles
    bool tracing_enabled = 5;
    bool metrics_enabled = 6;
    bool logging_enabled = 7;
    
    // Sampling
    double trace_sample_rate = 8;  // 0.0 to 1.0
    
    // Log level: TRACE, DEBUG, INFO, WARN, ERROR, FATAL
    string log_level = 9;
}
